### ISSUE TITLE
bug: Not support updating snis with PATCH

### ISSUE THREAD

**Issue by nanamikon** [apache/apisix#3351]
### Issue description
Fail to update snis using PATCH

### Environment

* apisix version: 2.2
* OS: ubuntu
* OpenResty / Nginx version:1.19

### Minimal test code / Steps to reproduce the issue

1.intsert ssl with snis 
2.patch snis

### What's the actual result? (including assertion message & call stack if applicable)
```
HTTP/1.1 400 Bad Request
Server: openresty
Date: Wed, 20 Jan 2021 01:20:37 GMT
Content-Type: application/json
Transfer-Encoding: chunked
Connection: keep-alive
Access-Control-Allow-Origin: *
Access-Control-Allow-Credentials: true
Access-Control-Expose-Headers: *
Access-Control-Max-Age: 3600

{"error_msg":"invalid configuration: additional properties forbidden, found 1"}
```

### What's the expected result?
200

**Comment by Firstsawyou** [comment]
Please check if there is a problem with your configuration. According to your error message, it is likely that your configuration is incorrect.

**Comment by spacewander** [comment]
> 1.intsert ssl with snis
> 2.patch snis

Could you show us how you send the requests?

**Comment by nanamikon** [comment]
I write a test case for this situation

```
=== TEST 1: not unwanted data, PUT
--- config
    location /t {
        content_by_lua_block {
            local json = require("toolkit.json")
            local t = require("lib.test_admin")

            local ssl_cert = t.read_file("t/certs/apisix.crt")
            local ssl_key =  t.read_file("t/certs/apisix.key")
            local data = {cert = ssl_cert, key = ssl_key, snis = {"not-unwanted-post.com"}}
            local code, message, res = t.test('/apisix/admin/ssl/1',
                ngx.HTTP_PUT,
                json.encode(data)
            )

            if code >= 300 then
                ngx.status = code
                ngx.say(message)
                return
            end

            res = json.decode(res)
            res.node.value.create_time = nil
            res.node.value.update_time = nil
            res.node.value.cert = ""
            res.node.value.key = ""
            res.node.key = nil
            ngx.say(json.encode(res))
        }
    }
--- response_body
{"action":"set","node":{"value":{"cert":"","id":"1","key":"","snis":["not-unwanted-post.com"],"status":1}}}


=== TEST 2: update snis, PATCH
--- config
    location /t {
        content_by_lua_block {
            local json = require("toolkit.json")
            local t = require("lib.test_admin")

            local data = {"update1.com", "update2.com"}
            local code, message, res = t.test('/apisix/admin/ssl/1/snis',
                ngx.HTTP_PATCH,
                json.encode(data)
            )

            if code >= 300 then
                ngx.status = code
                ngx.say(message)
                return
            end
        }
    }


```

**Comment by Firstsawyou** [comment]
> Test 2

The test case 2 should look like this:

![image](https://user-images.githubusercontent.com/52862365/105139087-c7963580-5b30-11eb-90ed-0e36f1b0c29e.png)

You try to modify it.

**Comment by nanamikon** [comment]
Thanks for reply,   so I can't update snis with sub path here ,  because the snis is array but not object

refer to https://github.com/apache/apisix/blob/master/doc/zh-cn/admin-api.md

sub path is not ok
...
# 替换路由的 upstream nodes -- sub path
$ curl http://127.0.0.1:9080/apisix/admin/routes/1/upstream/nodes -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PATCH -i -d '
{
    "39.97.63.200:80": 1
}'
HTTP/1.1 200 OK
```

This way can work well
``` 
# 替换路由的 methods -- 数组
$ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PATCH -i -d '{
    "methods": ["GET", "POST"]
}'
HTTP/1.1 200 OK
...

执行成功后，methods 将不保留原来的数据，整个更新为：
["GET", "POST"]
```

**Comment by Firstsawyou** [comment]
The `patch` method cannot use subpaths.
 By the way, the `local data = {"update1.com", "update2.com"}` data in your test case 2 is invalid, you need to be like test 1: `local data = {cert = ssl_cert, key = ssl_key, snis = {"not-unwanted-post.com"}}`

**Comment by nanamikon** [comment]
The patch method can use subpaths refer to 
![企业微信截图_16111308593158](https://user-images.githubusercontent.com/2010632/105146968-81929f00-5b3b-11eb-97a6-51246a3f9bfc.png)
and I have used subpaths in my project

Because I just want to update snis,  so the test case work fine for me
```
=== TEST 2: update snis, PATCH
--- config
    location /t {
        content_by_lua_block {
            local json = require("toolkit.json")
            local t = require("lib.test_admin")

            local data = {snis = {"update1.com", "update2.com"}}
            local code, message, res = t.test('/apisix/admin/ssl/1',
                ngx.HTTP_PATCH,
                json.encode(data)
            )

            if code >= 300 then
                ngx.status = code
                ngx.say(message)
                return
            end
        }
    }
```

**Comment by spacewander** [comment]
@nanamikon 
Should be fixed in #3405.

**Comment by nanamikon** [comment]
Thanks for reply,  I will use it in the next version of apisix
